test/test_search.py

```diff
@@ -66,5 +66,22 @@ def test_get_linearizer_actions(self):
     if Opt(OptOps.GROUPTOP, 0, 0) in actions:
       assert len([x for x in lins if x.applied_opts[0] == Opt(OptOps.GROUPTOP, axis=0, amt=3)]) == 0, "did not de-dup GROUPTOP"
 
+  def test_kernel_count(self):
+    """
+    Ensure that the kernel count is not incremented by _time_program when clearing l2
+    """
+    dev = Device[Device.DEFAULT]
+    assert dev.compiler is not None
+
+    si = [i for i in create_schedule([Tensor([1,2,3,4]).add(1).lazydata]) if i.ast[0].op not in LoadOps][0]
+    rawbufs = _ensure_buffer_alloc(bufs_from_lin(lin:=Linearizer(*si.ast)))
+    var_vals = {k:(k.max+k.min)//2 for k in lin.ast[0].vars()}
+    p = lin.to_program()
+
+    kernel_count = GlobalCounters.kernel_count
+    _time_program(p, dev.compiler.compile(p.src), var_vals, rawbufs,
```

Comment:
prefer not to test internal functions directly (function name that start with `_`). can you use `time_linearizer` instead? also kernel count not changing may due to nothing is being run, i'd test the count increased by exactly one after running one kernel

Reply:
I switched to testing `time_linearizer`. Though I don't understand the "running one kernel part". Is time_linearizer supposed to run kernels which count towards kcount ?

Reply:
it was 2 separate asks (1) don't call internal function in test, (2) test one kernel ran and the kernel count increased by 1 (not 7 or 10 in search).